src/loaders/map/resource.rs

```diff
@@ -17,3 +25,198 @@ impl ResourceType {
+                        let rotation = byte_stream.vector3();
+                        let scale = byte_stream.vector3();
+                        let transform = Transform::from(position, rotation.map(Deg), scale);
+                        let object = Object::new(None, model_name, None, transform);
+                        objects.push(object);
+                    }
+                }
+                ResourceType::LightSource => {
+                    let name = byte_stream.string(80);
+                    let position = byte_stream.vector3_flipped();
+                    let red = byte_stream.float32();
+                    let green = byte_stream.float32();
+                    let blue = byte_stream.float32();
+
+                    let color = Color::rgb_f32(red, green, blue);
+                    let range = byte_stream.float32();
+
+                    light_sources.push(LightSource::new(name, position, color, range));
+                }
+                ResourceType::SoundSource => {
+                    let name = byte_stream.string(80);
+                    let sound_file = byte_stream.string(80);
+                    let position = byte_stream.vector3_flipped();
+                    let volume = byte_stream.float32();
+                    let width = byte_stream.integer32();
+                    let height = byte_stream.integer32();
+                    let range = byte_stream.float32();
+
+                    let cycle = match byte_stream.get_version().equals_or_above(2, 0) {
+                        true => byte_stream.float32(),
+                        false => 4.0,
+                    };
+
+                    sound_sources.push(SoundSource::new(
+                        name,
+                        sound_file,
+                        position,
+                        volume,
+                        width as usize,
+                        height as usize,
+                        range,
+                        cycle,
+                    ));
+                }
+                ResourceType::EffectSource => {
+                    let name = byte_stream.string(80);
+                    let position = byte_stream.vector3_flipped();
+                    let effect_type = byte_stream.integer32();
+                    let emit_speed = byte_stream.float32();
+
+                    let _param0 = byte_stream.float32();
+                    let _param1 = byte_stream.float32();
+                    let _param2 = byte_stream.float32();
+                    let _param3 = byte_stream.float32();
+
+                    effect_sources.push(EffectSource::new(name, position, effect_type as usize, emit_speed));
+                }
+            }
+        }
+
+        Self {
+            resources_amount,
+            objects,
+            light_sources,
+            sound_sources,
+            effect_sources,
+        }
+    }
+}
+
+#[derive(Debug, PrototypeElement, new)]
```

Comment:
To do that we should do something with this [line](https://github.com/vE5li/korangar/blob/735b18ec2a29d186e9030824b29f2b8e6c934e98/procedural/src/byte/convertable.rs#L32) 
having all fields as Optionals (because they have version rules) throws an error there, something like "`Concat not implemented for [()]`"

Reply:
Hmm fair enough, I will try to fix that

Reply:
It is now fixed on master